broke-client/src/main/java/org/dsngroup/broke/client/BlockClient.java

```diff
@@ -28,15 +34,60 @@
 
     private int targetBrokerPort;
 
+    private Socket clientSocket;
+
+    private OutputStream clientOutputStream;
+
+    /**
+     * @param
+     */
     public void publish(String topic, int qos, int criticalOption, String payload) throws Exception {
-        Message msg = new Message(topic, qos, criticalOption, payload);
+        PublishMessage msg = (PublishMessage) MessageBuilder
+                .build("PUBLISH\r\nQoS:"+qos+",Critical-Option:"+criticalOption+",Topic:hello\r\n"+payload+"\r\n");
         // TODO: Send this message
+        clientOutputStream.write(("PUBLISH\r\nQoS:"+qos+",Critical-Option:"+criticalOption+",Topic:hello\r\n"+payload+"\r\n")
+                .getBytes(Charset.forName("UTF-8")));
+        // clientOutputStream.write(msg.toString().getBytes(Charset.forName("UTF-8")));
         System.out.println("Publish to topic: " + msg.getTopic() + "\nPayload: " + msg.getPayload());
     }
 
+    /**
+     * The default constructor
+     */
+    public BlockClient() {
+        this("0.0.0.0", 8181);
+    }
+
+    /**
+     * The constructor for creating a BlockClient
+     * 1. Build a connection to client
+     * 2. Send a CONNECT message to server.
+     * @param targetBrokerAddress The address of broker server.
+     * @param targetBrokerPort The port of broker server.
+     */
     public BlockClient(String targetBrokerAddress, int targetBrokerPort) {
         this.targetBrokerAddress = targetBrokerAddress;
         this.targetBrokerPort = targetBrokerPort;
         // TODO: Connect to the Broker Server, a.k.a. Send CONNECTION message
+        try {
+            clientSocket = new Socket();
+            clientSocket.setKeepAlive(true);
+            clientSocket.connect(new InetSocketAddress(this.targetBrokerAddress, this.targetBrokerPort));
+            clientOutputStream = clientSocket.getOutputStream();
+
+            clientOutputStream.write( "CONNECT\r\nQoS:0,Critical-Option:0\r\nnothing\r\n".getBytes(Charset.forName("UTF-8")) );
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    public void close() {
+        try {
+            clientSocket.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```

Comment:
Please mark each non-finished functionality into TODO, e.g. we should use `log` instead of `printStackTrace`.